x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/droppable.ts

```diff
@@ -84,15 +157,100 @@ const onReorderDrop = ({ columnId, setState, state, layerId, droppedItem }: Drop
   );
 
   return true;
-};
+}
+
+function onMoveDropToNonCompatibleGroup(props: DropHandlerProps<DraggedOperation>) {
+  const { columnId, setState, state, layerId, droppedItem } = props;
+
+  const layer = state.layers[layerId];
+  const op = { ...layer.columns[droppedItem.columnId] };
+  const field =
+    hasField(op) && state.indexPatterns[layer.indexPatternId].getFieldByName(op.sourceField);
+  if (!field) {
+    return false;
+  }
+
+  const operationSupportMatrix = getOperationSupportMatrix(props);
+  const operationsForNewField = operationSupportMatrix.operationByField[field.name];
+
+  if (!operationsForNewField || operationsForNewField.size === 0) {
+    return false;
+  }
+
+  const currentIndexPattern = state.indexPatterns[layer.indexPatternId];
+
+  const newLayer = insertOrReplaceColumn({
+    layer: deleteColumn({
+      layer,
+      columnId: droppedItem.columnId,
+      indexPattern: currentIndexPattern,
+    }),
+    columnId,
+    indexPattern: currentIndexPattern,
+    op: operationsForNewField.values().next().value,
+    field,
+  });
+
+  trackUiEvent('drop_onto_dimension');
+  const hasData = Object.values(state.layers).some(({ columns }) => columns.length);
+  trackUiEvent(hasData ? 'drop_non_empty' : 'drop_empty');
```

Comment:
I notice that you've added these events, but they were really intended for use in the workspace panel- we can safely remove this line.